### PR TITLE
(1) mobile:  rewrap settings to use sections.

### PR DESCRIPTION
**Pull request by janiversen**
### Describe the pull request:

- [ ] Bug fix
- [ ] Functional change
- [ x] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This PR changes settings.qml to use sections, this allows much more real estate for each
part, and thus more information is possible.

See image of how settings look when first opened
initialSettings

And when opening e.g. Theme:
openTheme

In one of the next PRs. Units will be expanded to allow "personalised" as in the desktop version.

In one of the next PRs. Theme will be expanded to allow user selected colours as well as day/night shift.

Which feature belong in which section is surely something, that can be talked about for a long time, and I have no problem moving features around.

The medium term plan is to make nearly all settings from the desktop available.

### Changes made:

commit 1: merge of planner 1-4 this needs to be removed
commit 2-4: new templates
commit 5: bump versions in settings to reflect qt5.12 in lower limit
remains: integrate TemplateSections

### CODE REVIEW

**Comment by dirkhh** [comment]
I like the new look that you show in those screen shots.

**Comment by janiversen** [comment]
Ready to be reviewed and merged.

File: mobile-widgets/qml/TemplateLine.qml
Comment (dirkhh) [inline mobile-widgets/qml/TemplateLine.qml]: absolute offsets tend to be a problem - we have users using this on screens that vary widely in width. So 20 pixels could be almost 10% of the width, or only 1% of the width. We try to express width in terms of `Kirigami.Units.gridUnit` (or `...smallSpacing` and `...largeSpacing` - these are all multiples of each other)

File: mobile-widgets/qml/TemplateTitle.qml
Comment (dirkhh) [inline mobile-widgets/qml/TemplateTitle.qml]: same comment about using `Kirigami.Units`

File: mobile-widgets/qml/TemplateTitle.qml
Comment (dirkhh) [inline mobile-widgets/qml/TemplateTitle.qml]: I realize that a user of the template should overwrite the text... so if you want to use the text set here to remind the developer of that, maybe have a slightly more instructive text `"using the TemplateTilte you need to set the title text"`

**Review by dirkhh — APPROVED** [review]
Really small comments. To avoid another round and wasting your time, I'll try to adjust this as I merge it. If you think I got something wrong or would like to do things differently, please send an incremental PR after this is merged.

File: mobile-widgets/qml/Settings.qml
Comment (dirkhh) [inline mobile-widgets/qml/Settings.qml]: funny thing... your commit messages says "without changing anything", but here an `Enums` change snuck in, which depends on the other PR that is still WIP... :-(

**Comment by dirkhh** [comment]
I have two options:
- I can quickly clean this up to work on top of current master, but then your WIP PRs dealing with your larger infrastructure changes for Enums need to include the `CloudStatus.CS_NOCLOUD` -> `Enums.CS_NOCLOUD` (and others) changes in these files here as well
- I can hold off until the WIP is ready to be merged

What's your preference?

**edited to add: never mind, see below, there are more dependencies on other, pending PRs**

**Review by dirkhh — CHANGES_REQUESTED** [review]
I can still do the simple changes that I mentioned earlier, so this isn't really "requesting changes" from you... this just is intended to change the status from 'approved' to 'not ready to be merged'

File: mobile-widgets/qml/Settings.qml
Comment (janiversen) [reply mobile-widgets/qml/Settings.qml]: Sorry seems something got wrong with my rebasing, locally I have corrected most of this. will force push a bit later

Comment (dirkhh) [reply mobile-widgets/qml/Settings.qml]: no worry, this is easy to happen given how many branches you have in flight that all touch the same files.
Would it be easier to try and get the enum and theme changes sorted so we can get that merged?
I don't really want to call you additional work to change things back and forth, especially since I expect we will in the end get all of these parts merged, right?

File: mobile-widgets/qml/TemplateLine.qml
Comment (janiversen) [reply mobile-widgets/qml/TemplateLine.qml]: I think I got it. However the -20 was a real bug, width = parent.width. But I will try to remember this for future changes.

File: mobile-widgets/qml/TemplateTitle.qml
Comment (janiversen) [reply mobile-widgets/qml/TemplateTitle.qml]: I corrected this one as well (problem is when we start to use, different layout for e.g. iPad and iPhone), this was my fast way of adding a width.

I have been looking for an example showing width = max width - a bit for padding. If you have an example please point me to it.

File: mobile-widgets/qml/TemplateTitle.qml
Comment (janiversen) [reply mobile-widgets/qml/TemplateTitle.qml]: changed.

File: mobile-widgets/qml/Settings.qml
Comment (janiversen) [reply mobile-widgets/qml/Settings.qml]: actually it does not depend on the WIP theming, subsurfaceTheme is the good old one.

**Comment by janiversen** [comment]
The only problem, was in the cloud setting, where I used Enums. instead of the old CloudStatus, that is corrected.

I am still unsure how to handle the width. What I need often if max. width (== parent.width) - a small bit.

Looking forward to read any new review comments from you. If merged, this PR calls for documentation changes, but I am too unsure how to signal that.

**Comment by dirkhh** [comment]
On my phone, sitting in traffic.
This removes the calls to blueTheme() and corresponding functions that are needed in the QML version to change themes.
Or am I missing some subtle change that makes those no longer necessary?

On January 18, 2020 1:14:12 PM PST, jan iversen <notifications@github.com> wrote:
>janiversen commented on this pull request.
>
>
>
>> -			}
>-			SsrfSwitch {
>-				id: blueButton
>-				Layout.preferredWidth: gridWidth * 0.25
>-				checked: subsurfaceTheme.currentTheme === "Blue"
>-				enabled: subsurfaceTheme.currentTheme !== "Blue"
>-				onClicked: {
>-					blueTheme()
>-					PrefDisplay.theme = subsurfaceTheme.currentTheme
>+				SsrfSwitch {
>+					id: blueButton
>+					Layout.preferredWidth: gridWidth * 0.25
>+					checked: subsurfaceTheme.currentTheme === "Blue"
>+					enabled: subsurfaceTheme.currentTheme !== "Blue"
>+					onClicked: {
>+						subsurfaceTheme.currentTheme = "Blue"
>
>actually it does not depend on the WIP theming, subsurfaceTheme is the
>good old one.
>
>-- 
>You are receiving this because you commented.
>Reply to this email directly or view it on GitHub:
>https://github.com/Subsurface-divelog/subsurface/pull/2537#discussion_r368248902

-- 
From my phone

**Comment by janiversen** [comment]
How can I have overlooked that !! clearly an error, only excuse is that QML did not complain.

Traffic jam is not very usual here, unless more than 3 cars try to pass (within an hour) :-)

**Comment by janiversen** [comment]
putting it back to WIP status, I am not 100% sure to catch all changes, so I prefer this PR waits until we have agreed on foundation1. Once that is agreed upon a lot of things will be easier for me (which is why it is called foundation1).

**Comment by janiversen** [comment]
I have now manually compared each gridLayout before -> after, and see no differences. BUT if I am allowed to ask a favor, I would really appreciate if you control the gridLayout as well.

In the commits where gridLayouts are moved, it should be a pure move (apart from the initial text, which becomes section title), changes are wrong.

Once  this PR is accepted I will submit the last part of foundation1 and then update the rest.

**Comment by dirkhh** [comment]
merged locally to do some small commit message edits. otherwise it looked perfect and I like the new look.

**Pull request closed without merging**
